cmd/gardener-extension-provider-local/app/app.go

```diff
@@ -251,6 +252,11 @@ func NewControllerManagerCommand(ctx context.Context) *cobra.Command {
 			reconcileOpts.Completed().Apply(&oscommon.DefaultAddOptions.IgnoreOperationAnnotation)
 			reconcileOpts.Completed().Apply(&localworker.DefaultAddOptions.IgnoreOperationAnnotation)
 
+			// TODO(rfranzke): Remove the ManageMCM fields as soon as the general options no longer support the
+			//  GardenletManagesMCM field.
+			localworker.DefaultAddOptions.ManageMCM = !generalOpts.Completed().GardenletManagesMCM
+			controlplanewebhook.ManageMCM = !generalOpts.Completed().GardenletManagesMCM
```

Comment:
From the controlplane webhook POV, should it not be:
```
			controlplanewebhook.ManageMCM = generalOpts.Completed().GardenletManagesMCM
```
Since the controlplane webhook manages MCM if gardenlet manages MCM.

Reply:
😄 Yeah, you could argue that way, although it is a bit confusing as well (while the current approach is also confusing, obviously). Nevertheless, I can change it when marking this PR as ready.

Reply:
Why not name the variable `GardenletManagesMCM` itself?

Reply:
Good suggestion, will improve this eventually 👍🏻